vault/mount.go

```diff
@@ -972,9 +976,11 @@ func (c *Core) taintMountEntry(ctx context.Context, nsID, mountPath string, upda
 // * PendingRemoval - log an error about builtin deprecation and return an error
 // if VAULT_ALLOW_PENDING_REMOVAL_MOUNTS is unset
 // * Removed - log an error about builtin deprecation and return an error
-func (c *Core) handleDeprecatedMountEntry(ctx context.Context, entry *MountEntry, pluginType consts.PluginType) (*logical.Response, error) {
+func (c *Core) handleDeprecatedMountEntry(ctx context.Context, entry *MountEntry, pluginType consts.PluginType) error {
```

Comment:
This func is a bit confusing now, because it has one explicit return value (error) and one implicit one (ctx).  What makes it worse is that error and ctx are ubiquitous so the surprise is heightened.  

I've never liked it when we had to resort to stuffing things into ctxs, but this is a little weirder because usually when a func does that, it's deriving a new ctx with the new field, and it's only impacting its call graph.  Whereas here it's intended as a side effect.  Does it really work?  Yes, interfaces, are pointers, but we're not mutating the contents of the ctx, we're just assigning to the variable/param holding that pointer.

Reply:
Unless my testing was incorrect, this works. I also am not a fan of this approach, but it was the only solution I could think of to achieve what we wanted to.

The justification for doing this is we'd either need to have error-handling logic at multiple levels of the stack, or else find a way to package up errors for later processing. Since context-stuffing is already a practice, I decided to use it.

I am open to other suggestions for approaches here. At the very least, I can split this out into two functions (one which has side-effects, and one that is side-effect-free). OR I can just add a comment to the function (which I should have done to begin with).

Barring some other solution for this, I think I'd prefer to leave it as-is. As it stands, we _do_ stuff values into contexts